src/why3prop.ml

```diff
@@ -0,0 +1,127 @@
+(** Translation of lambdapi goals to why3 goals in propositional case *)
+
+open Terms
+open Extra
+
+exception NoGoalTranslation
+
+(* a map from lambdapi terms to Why3 constants *)
+type cnst_table = (term * Why3.Term.lsymbol) list
+
+(* a map from labels to Why3 terms. *)
+type ctxt_labels = (string * Why3.Term.term) list
+
+(** [get_newname ()] generates a new axiom name. *)
+let get_newname : unit -> string =
+    (* number of axioms proved whith the Why3 tactic *)
+    let nbr_axioms : int ref = ref 0 in fun () ->
+    nbr_axioms := !nbr_axioms + 1;
+    "Why3axiom_" ^ (string_of_int !nbr_axioms)
+
+(* builtins configuration for propositional logic *)
+type prop_config =
+  { symb_P     : sym (** Encoding of propositions.        *)
+  ; symb_T     : sym (** Encoding of types.               *)
+  ; symb_or    : sym (** Disjunction(∨) symbol.           *)
+  ; symb_and   : sym (** Conjunction(∧) symbol.           *)
+  ; symb_imp   : sym (** Implication(⇒) symbol.           *)
+  ; symb_bot   : sym (** Bot(⊥) symbol.                   *)
+  ; symb_not   : sym (** Not(¬) symbol.                   *) }
+
+(** [get_prop_config pos builtins] set the builtins configuration using
+    [builtins] *)
+let get_prop_config :
+    Pos.popt -> sym StrMap.t -> prop_config = fun pos builtins ->
+    let find_sym key =
+        try StrMap.find key builtins with Not_found ->
+        Console.fatal pos "Builtin symbol [%s] undefined." key
+    in
+    { symb_P     = find_sym "P"
+    ; symb_T     = find_sym "T"
+    ; symb_or    = find_sym "or"
+    ; symb_and   = find_sym "and"
+    ; symb_imp   = find_sym "imp"
+    ; symb_bot   = find_sym "bot"
+    ; symb_not   = find_sym "not" }
+
+(** [translate pos builtins (hs, g)] translate from lambdapi to Why3 goal [g]
```

Comment:
English: translate -> translates, output -> outputs

Suggestion:
[translate pos builtins (hs, g)] returns [(tmap,smap,t)] where:
- [t] is a Why3 term representing the Lambdapi goal [(hs,g)],
- [tmap] maps abstracted Lambdapi terms to Why3 constants,
- [smap] maps ...

Reply:
Please explain what is smap.

Reply:
Done in c363bd5e1842c2985e8afee8ce8a2f4201468562.